R/graphql-builders.R

```diff
@@ -0,0 +1,155 @@
+#' S7 class for representing GraphQL query configurations
+#' @keywords internal
+#' @noRd
+meetup_template <- S7::new_class(
+  "meetup_template",
+  properties = list(
+    template = S7::class_character,
+    page_info_path = S7::class_character,
+    edges_path = S7::class_character,
+    process_data = S7::class_function
+  )
+)
+
+#' Constructor for meetup_template
+#' @keywords internal
+#' @noRd
+meetup_template_query <- function(
+  template,
+  page_info_path,
+  edges_path,
+  process_data = process_graphql_list
+) {
+  meetup_template(
+    template = template,
+    page_info_path = page_info_path,
+    edges_path = edges_path,
+    process_data = process_data
+  )
+}
+
+#' Execute a meetup_template
+#' @keywords internal
+#' @noRd
+execute <- S7::new_generic("execute", "object")
+
+S7::method(execute, meetup_template) <- function(
+  object,
+  ...,
+  max_results = NULL,
+  handle_multiples = "list",
+  extra_graphql = NULL,
+  .progress = TRUE
+) {
+  all_data <- list()
+  cursor <- NULL
+  previous_cursor <- NULL
+  page_count <- 0
+
+  repeat {
```

Comment:
Not often I see this!

Reply:
totally Ben's code! I didnt even know it existed, and am not entirely sure how  it works :P But its a clean solution!